convolution no longer compatible with scipy 1.5.0 (Windows)

```
============================= test session starts =============================
platform win32 -- Python 3.8.5, pytest-6.2.3, py-1.9.0, pluggy-0.13.1

Running tests with Astropy version 4.3.dev1242+g9f1b7775b.
Running tests in astropy/convolution/tests/test_convolve_nddata.py.

Date: 2021-04-27T16:14:25

Platform: Windows-10-10.0.19041-SP0

Executable: c:\...\miniconda3\envs\py38\python.exe

Full Python Version:
3.8.5 | packaged by conda-forge | (default, Jul 31 2020, 01:53:45) [MSC v.1916 64 bit (AMD64)]

encodings: sys: utf-8, locale: cp1252, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions:
Numpy: 1.19.1
Scipy: 1.5.0
Matplotlib: 3.3.3
h5py: 2.10.0
Pandas: 1.1.2
PyERFA: 1.7.2
Cython: 0.29.21
Scikit-image: not available
asdf: 2.7.1

Using Astropy options: remote_data: none.

ARCH_ON_CI: undefined
IS_CRON: undefined

rootdir: D:\...\astropy, configfile: setup.cfg
plugins: asdf-2.7.1, hypothesis-5.29.4, arraydiff-0.3, astropy-header-0.1.2, cov-2.10.1, doctestplus-0.9.0, filter-subpackage-0.1.1, forked-1.3.0, openfiles-0.5.0, remotedata-0.3.2, xdist-2.1.0
collected 3 items

astropy\convolution\tests\test_convolve_nddata.py F.F                    [100%]

================================== FAILURES ===================================
______________________________ test_basic_nddata ______________________________

    def test_basic_nddata():
        arr = np.zeros((11, 11))
        arr[5, 5] = 1
        ndd = NDData(arr)
        test_kernel = Gaussian2DKernel(1)

        result = convolve(ndd, test_kernel)

        x, y = np.mgrid[:11, :11]
        expected = result[5, 5] * np.exp(-0.5 * ((x - 5)**2 + (y - 5)**2))

        np.testing.assert_allclose(result, expected, atol=1e-6)

>       resultf = convolve_fft(ndd, test_kernel)

astropy\convolution\tests\test_convolve_nddata.py:24:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astropy\nddata\decorators.py:246: in wrapper
    result = func(data, *args, **kwargs)
astropy\convolution\convolve.py:731: in convolve_fft
    newshape = _next_fast_lengths(newshape)
astropy\convolution\convolve.py:85: in _next_fast_lengths
    return np.array([scipy.fft.next_fast_len(j, real=False) for j in shape])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

.0 = <iterator object at 0x000001EED13E6310>

>   return np.array([scipy.fft.next_fast_len(j, real=False) for j in shape])
E   TypeError: good_size() takes no keyword arguments

astropy\convolution\convolve.py:85: TypeError
________________________ test_masked_nddata[<lambda>1] ________________________

convfunc = <function <lambda> at 0x000001EED13CB3A0>

    @pytest.mark.parametrize('convfunc',
       [lambda *args: convolve(*args, nan_treatment='interpolate', normalize_kernel=True),
        lambda *args: convolve_fft(*args, nan_treatment='interpolate', normalize_kernel=True)])
    def test_masked_nddata(convfunc):
        arr = np.zeros((11, 11))
        arr[4, 5] = arr[6, 5] = arr[5, 4] = arr[5, 6] = 0.2
        arr[5, 5] = 1.5
        ndd_base = NDData(arr)

        mask = arr < 0  # this is all False
        mask[5, 5] = True
        ndd_mask = NDData(arr, mask=mask)

        arrnan = arr.copy()
        arrnan[5, 5] = np.nan
        ndd_nan = NDData(arrnan)

        test_kernel = Gaussian2DKernel(1)

>       result_base = convfunc(ndd_base, test_kernel)

astropy\convolution\tests\test_convolve_nddata.py:47:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astropy\convolution\tests\test_convolve_nddata.py:30: in <lambda>
    ???
astropy\nddata\decorators.py:246: in wrapper
    result = func(data, *args, **kwargs)
astropy\convolution\convolve.py:731: in convolve_fft
    newshape = _next_fast_lengths(newshape)
astropy\convolution\convolve.py:85: in _next_fast_lengths
    return np.array([scipy.fft.next_fast_len(j, real=False) for j in shape])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

.0 = <iterator object at 0x000001EED14500A0>

>   return np.array([scipy.fft.next_fast_len(j, real=False) for j in shape])
E   TypeError: good_size() takes no keyword arguments

astropy\convolution\convolve.py:85: TypeError
=========================== short test summary info ===========================
FAILED astropy/convolution/tests/test_convolve_nddata.py::test_basic_nddata
FAILED astropy/convolution/tests/test_convolve_nddata.py::test_masked_nddata[<lambda>1]
========================= 2 failed, 1 passed in 1.88s =========================
```

The tests passed when I upgraded `scipy` to 1.6.3. We currently have `scipy` 1.1 as the minversion.

https://github.com/astropy/astropy/blob/298c8e9a2e25bfeeda12762c700aa58d44ff7b30/setup.cfg#L66

Why didn't we catch this on Linux?

https://github.com/astropy/astropy/blob/298c8e9a2e25bfeeda12762c700aa58d44ff7b30/.github/workflows/ci_workflows.yml#L70

Could this be due to #11533 , @dhomeier ? cc @keflavich and @WilliamJamieson

--- a/astropy/convolution/convolve.py
+++ b/astropy/convolution/convolve.py
@@ -85,6 +85,11 @@ def _next_fast_lengths(shape):
         return np.array([scipy.fft.next_fast_len(j, real=False) for j in shape])
     except ImportError:
         pass
+    except TypeError as ex:
+        # this error happens with scipy version <=1.5
+        if "good_size() takes no keyword arguments" not in str(ex):
+            raise ex
+
 
     newshape = np.empty(len(np.atleast_1d(shape)), dtype=int)
     for i, j in enumerate(shape):

--- a/astropy/convolution/convolve.py
+++ b/astropy/convolution/convolve.py
@@ -82,13 +82,9 @@ def _next_fast_lengths(shape):
 
     try:
         import scipy.fft
-        return np.array([scipy.fft.next_fast_len(j, real=False) for j in shape])
+        return np.array([scipy.fft.next_fast_len(j) for j in shape])
     except ImportError:
         pass
-    except TypeError as ex:
-        # this error happens with scipy version <=1.5
-        if "good_size() takes no keyword arguments" not in str(ex):
-            raise ex
 
     newshape = np.empty(len(np.atleast_1d(shape)), dtype=int)
     for i, j in enumerate(shape):

--- a/astropy/convolution/convolve.py
+++ b/astropy/convolution/convolve.py
@@ -90,7 +90,6 @@ def _next_fast_lengths(shape):
         if "good_size() takes no keyword arguments" not in str(ex):
             raise ex
 
-
     newshape = np.empty(len(np.atleast_1d(shape)), dtype=int)
     for i, j in enumerate(shape):
         scale = 10 ** max(int(np.ceil(np.log10(j))) - _good_range, 0)

